avocado/utils/dmesg_utils.py

```diff
@@ -0,0 +1,121 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2020 IBM
+# Author: Praveen K Pandey <praveen@linux.vnet.ibm.com>
+#
+
+"""
+Module for manipulate dmesg while running test.
+"""
+
+
+import time
+import os
+import tempfile
+import logging
+
+from . import process, genio
+
+LOGGER = logging.getLogger('avocado.test')
+
+
+class DmesgError(Exception):
+    """
+    Base Exception Class for all dmesg  utils exceptions
+    """
+    pass
+
+
+def clear_dmesg():
+    """
+    This function clear dmesg of system
+    and save older dmesg in a file
+    """
+
+    cmd = "dmesg -c"
+    status = process.system(cmd, timeout=30, ignore_status=True,
+                            verbose=False, shell=True, sudo=True)
+    if status:
+        raise DmesgError(
+            "unable to clear dmesg as some issue while clearing")
+
+
+def collect_dmesg(dmesg_log_file=None):
+    """
+    This Function collect dmesg and save in a tmp file
+    :return: file which contain dmesg
+    """
+    if dmesg_log_file is None:
+        dmesg_log_file = tempfile.mktemp(
+            suffix="-%s" % time.strftime("%Y-%m-%d:%H:%M:%S"), dir=tempfile.gettempdir())
+    dmesg = process.system_output(
+        "dmesg", ignore_status=True, sudo=True).decode()
+    genio.write_file(dmesg_log_file, dmesg)
+    if not os.path.isfile(dmesg_log_file):
+        raise DmesgError("issue with dmesg store in file")
+    return dmesg_log_file
+
+
+def verify_patterns_dmesg(dmesg_log_file=None, pattern=None):
```

Comment:
ok, why we have this as `verify_patterns_dmesg` as we can pass a list of pattern to function as to why call traces I made as default because in our test we use this most of the cases, I will enhance docstring

Reply:
if `pattern` can accept a list, then we should rename it to `patterns`